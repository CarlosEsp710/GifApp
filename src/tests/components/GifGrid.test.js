import React from "react";
import { shallow } from "enzyme";

import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("GifGrid", () => {
  const category = "Hannibal";

  test("should return component <GifGrid/>", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should return <GifGridItem/> list when data is loaded from useFetchGifs", () => {
    const images = [
      {
        id: "abc",
        url: "http://example.com/categoryName/image.jpg",
        title: "Some thing",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: images,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should not return the loading when data is loaded", () => {
    const images = [
      {
        id: "abc",
        url: "http://example.com/categoryName/image.jpg",
        title: "Some thing",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: images,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(images.length);
  });
});
