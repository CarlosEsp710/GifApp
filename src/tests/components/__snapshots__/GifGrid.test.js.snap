// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid should return <GifGridItem/> list when data is loaded from useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-1s"
  >
    Hannibal
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Some thing"
      url="http://example.com/categoryName/image.jpg"
    />
  </div>
</Fragment>
`;

exports[`GifGrid should return component <GifGrid/> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn animate__delay-1s"
  >
    Hannibal
  </h3>
  <div
    className="card-grid"
  >
    <p
      className="animate__animated animate__flash"
    >
      Loading...
    </p>
  </div>
</Fragment>
`;
